transformer/src/testTable.ts

```diff
@@ -0,0 +1,195 @@
+    instanciation,
+    iValue,
+  };
+}
+
+/**
+ * Checks replacer.
+ *
+ * Replace checksThatThe and checksForEachLineThatThe function calls with
+ * assemblyscript code that use unittest functions.
+ */
+export class TestTable {
+  static strPattern = 'checksForEachLineThatThe';
+
+  /**
+   * Replaces all checks functions.
+   *
+   * XXX: this function is clearly too big. Feel free to split it.
+   *
+   * @param {Node} node
+   * @return {Node}
+   */
+  static transform(node: CallExpression): Expression {
+    // Push all the arguments of the checksForEachLineThatThe function call to args array.
+    // The function signature is the following:
+    // testTable(<test set name>, <template of value to compute>,
+    // <comparison criterion>, <template of expected result>, <test failure strategy>, [<test values>...])
+    // <template of expected result>> field is optional
+    const args: string[] = [];
+    node.args.forEach((element) => {
+      const content = element.range.source.text.slice(
+        element.range.start,
+        element.range.end,
+      );
+      args.push(content);
+    });
+
+    const testSetName = args[0];
+    const gotTemplate = args[1]!;
+    const comparisonCriterion = args[2]!;
+    let expectedTemplate = args.length == 5 ? args[3]! : '';
+
+    // extracts all test values and converts potential breaking single quotes to double ones.
+    const rawValues = (node.args[args.length - 1] as ArrayLiteralExpression)
+      .elementExpressions;
+    const values = rawValues.map((e) =>
+      e.range.source.text.slice(e.range.start, e.range.end).replace(/'/g, '"'),
+    );
+
+    let expr = `describe(${testSetName}, () => {\n`;
+
+    const {ifExpr, hasWant: needWant} =
+      generateIfExpression(comparisonCriterion);
+
+    if (needWant && expectedTemplate == '') {
+      expectedTemplate = 'arg0';
+    }
+
+    let gotExpr;
+    let expectExpr;
+
+    // two concepts here:
+    // - testCounter keeps track of the number of tests
+    // - iValue is the index of the values array
+    // One test can involve one or multiple values.
+    // Here we loop until all test values are used.
+    let testCounter = 0;
+    for (let iValue = 0; iValue < values.length; testCounter++) {
+      // Destructuring assignment unpacks returned object values into distinct values.
+      ({instanciation: gotExpr, iValue} = hydrateTemplate(
+        gotTemplate,
+        values,
+        iValue,
+      ));
+      ({instanciation: expectExpr, iValue} = hydrateTemplate(
+        expectedTemplate,
+        values,
+        iValue,
+      ));
+
+      const test = generateTest(
+        testCounter.toString(), // use test counter as test name
+        gotExpr.replace(/['`]/g, ''),
+        expectExpr.replace(/['`]/g, ''),
+        ifExpr,
+        needWant,
+      );
+
+      expr += test.replace(/\n/g, '\n  '); // adds static indentation
+    }
+    expr += `\n});`;
+
+    const processedNode = SimpleParser.parseExpression(expr);
```

Comment:
Your terminology is misleading: your are creating a new node and the processedNode is the current node variable.